testsuite/features/step_definitions/vm_steps.rb

```diff
@@ -0,0 +1,410 @@
+# Copyright (c) 2023 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### This file contains all steps concerning virtual machine management
+
+When(/^I create a (leap|sles|rhlike|deblike) virtual machine named "([^"]*)" (with|without) cloudinit on "([^"]*)"$/) do |os_type, vm_name, cloudinit, host|
+  node = get_target(host)
+  disk_path = "/tmp/#{vm_name}_disk.qcow2"
+
+  # The cloud init images are generated via sumaform and can be modified there
+  # The names of the qcow2 images are also defined there
+  cloudinit_path = "/tmp/cloudinit-disk-#{os_type}.iso"
+  memory = '512'
+
+  case os_type
+  when 'leap'
+    name = 'leap-disk-image-template.qcow2'
+    net =
+      if cloudinit == 'with'
+        'salt-leap'
+      else
+        'test-net0'
+      end
+    os = 'opensuse15.4'
+    memory = '1024'
+    cloudinit_image = "/var/testsuite-data/cloudinit-disk-#{os_type}.iso"
```

Comment:
The assignment of `cloudinit_image`, we don't need to repeat it in all the cases.
We can extract this **before** the case statement, and only if in some specific case the string format must be different, then we override it inside the case statement.

Reply:
Adjusted.